sys/userns_linux.go

```diff
@@ -0,0 +1,53 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package sys
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+)
+
+// RunningInUserNS detects whether we are currently running in a user namespace.
+// Originally copied from github.com/lxc/lxd/shared/util.go
+func RunningInUserNS() bool {
```

Comment:
We should add `sync.Once`, but can be another PR.

Reply:
Was actually just looking at that; I noticed we did it elsewhere, so I started writing some code to move it here (this PR still is a draft, so I could make some changes (waiting for https://github.com/containerd/containerd/pull/4227)

Reply:
Pushed a commit to use `sync.Once`

Reply:
thanks